ui/raidboss/data/triggers/dun_scaith.js

```diff
@@ -0,0 +1,483 @@
+      regex: /03:........:Added new combatant Aether Collector/,
+      suppressSeconds: 5,
+      alertText: {
+        en: 'Kill collectors',
+      },
+    },
+
+
+    // SCATHACH
+
+    {
+      // The actual attack is 1D20, but the castbar windup is 1D1F
+      id: 'Dun Scaith Shadespin',
+      regex: / 14:1D1(E|F):Scathach starts using Shadespin/,
+      suppressSeconds: 5,
+      alertText: {
+        en: 'Avoid arm slaps',
+      },
+    },
+    {
+      id: 'Dun Scaith Thirty Thorns',
+      regex: / (1[56]:........:Scathach:1D2B:Thirty Thorns|1[56]:........:Scathach:1D1B:Soar)/,
+      suppressSeconds: 5,
+      alertText: {
+        en: 'Out of melee',
+      },
+    },
+    {
+      id: 'Dun Scaith Thirty Arrows',
+      regex: / 14:1D2F:Scathach starts using Thirty Arrows/,
+      alertText: {
+        en: 'Avoid line AoEs',
+      },
+    },
+    {
+      id: 'Dun Scaith Thirty Souls',
+      regex: / 14:1D32:Scathach starts using Thirty Souls/,
+      condition: function(data) {
+        return data.role == 'healer';
+      },
+      alertText: {
+        en: 'Raid damage',
+      },
+    },
+    {
+      // Ordinarily we wouldn't use a game log line for this.
+      // However, the RP text seems to be the only indicator.
+      id: 'Dun Scaith Shadow Links',
+      regex: /Shadows gather on the floor/,
+      suppressSeconds: 5,
+      alertText: {
+        en: 'Stop moving',
+      },
+    },
+    {
+      id: 'Dun Scaith Shadow Limb Spawn',
+      regex: / 03:........:Added new combatant Shadow Limb/,
+      suppressSeconds: 5,
+      alertText: {
+        en: 'Kill the hands',
+      },
+    },
+    {
+      id: 'Dun Scaith Connla Spawn',
+      regex: / 14:1CD1:Connla starts using Pitfall/,
+      alertText: {
+        en: 'Avoid AoE, Kill Connla',
+      },
+    },
+
+    // These triggers are common to both Scathach and Diabolos
+
+    {
+      id: 'Dun Scaith Nox Orbs',
+      regex: / 1B:........:(\y{Name}):....:....:005C/,
+      suppressSeconds: 5,
+      alertText: function(data, matches) {
+        if (matches[1] == data.me) {
+          return {
+            en: 'Take orb outside',
+          };
+        }
+      },
+    },
+    {
+      id: 'Dun Scaith Shadethrust',
+      regex: / 14:(?:1D23:Scathach|1C1A:Diabolos Hollow) starts using Shadethrust/,
+      alertText: {
+        en: 'Away from front',
+      },
+    },
+
+    // DIABOLOS
+
+    {
+      id: 'Dun Scaith Ultimate Terror',
+      regex: / 14:1C12:Diabolos starts using Ultimate Terror/,
+      alertText: {
+        en: 'Get in',
+      },
+    },
+    {
+      id: 'Dun Scaith Nightmare',
+      regex: / 14:(1C0E|1C20):\y{Name} starts using (Nightmare|Hollow Nightmare)/,
+      alertText: {
+        en: ' Look away',
+      },
+    },
+    {
+      id: 'Dun Scaith Noctoshield',
+      regex: / 1A:........:Diabolos (?:gains|loses) the effect of Noctoshield/,
+      condition: function(data) {
+        return data.role != 'dps';
```

Comment:
If the divisions before were specifically to determine who could do what crowd control, I agree that there's no reason to retain the division. I can't think of any mechanics where Cactbot would have to call them differently based on whether the player is melee vs not. Can[ACTION]() is a much better way to handle it, yes.

Reply:
Oh! I think maybe the issue here is that it's `(?:gains|loses)` and not `(gains|loses)`.  I think you mean the latter, since you're using the match later.  That's probably why there's two.  One for start, and one for end.

Reply:
... The forgetful match would definitely be it. I appreciate it, this is why I wanted at least one other person to have a look.